server/cluster_info.go

```diff
@@ -78,6 +79,45 @@ func loadClusterInfo(id core.IDAllocator, kv *core.KV, opt *scheduleOption) (*cl
 	return c, nil
 }
 
+func (c *clusterInfo) OnChangeClusterVersion() {
+	var (
+		minVersion     semver.Version
+		clusterVersion semver.Version
+	)
+
+	clusterVersion = c.opt.loadClusterVersion()
+	stores := c.GetStores()
+	for i, s := range stores {
+		if s.IsTombstone() {
+			continue
+		}
+		if i == 0 {
```

Comment:
I think it would be clearer to move this to after the `v` assignment as `if i==0 || v.LessThan(minVersion)`.